miniapp/kernel/miniapp_larft_gemv.cpp

```diff
@@ -0,0 +1,230 @@
+    const SizeType ldv = opts.ldv;
+    const SizeType ldt = opts.ldt;
+
+    getter_random<T> random_value(25698);
+    auto rnd = [&random_value](const TileElementIndex&) { return random_value(); };
+    auto tau = createTile<T, Device::CPU>(rnd, {k, 1}, k);
+    auto v = createTile<T, Device::CPU>(rnd, {m, k}, ldv);
+
+    // As the kernels need T to be zeroed we start from there.
+    auto el_t = [](const TileElementIndex&) { return T{0.}; };
+
+    WorkTiles<T, device> vs(opts.count, m, k, ldv);
+    WorkTiles<T, device> ts(opts.count, k, k, ldt);
+    WorkTiles<T, device> taus(opts.count, k, 1, k);
+
+    vs.setElementsFromTile(v);
+    ts.setElements(el_t);
+
+    [[maybe_unused]] auto kernel_MC = [m, k, &vs, &tau, &ts](SizeType i) {
+      MC_reference(m, k, vs(i).ptr(), vs(i).ld(), tau.ptr(), ts(i).ptr(), ts(i).ld());
+    };
+#ifdef DLAF_WITH_CUDA
+    [[maybe_unused]] auto kernel_GPU0 = [m, k, &vs, &tau, &ts](SizeType i, cublasHandle_t handle) {
+      gpulapack::larft_gemv0(handle, m, k, vs(i).ptr(), vs(i).ld(), tau.ptr(), ts(i).ptr(), ts(i).ld());
+    };
+
+    [[maybe_unused]] auto copy_tau_in_t = [k, &tau, &ts](SizeType i, whip::stream_t stream) {
+      gpulapack::lacpy(blas::Uplo::General, 1, k, tau.ptr(), 1, ts(i).ptr(), ts(i).ld() + 1, stream);
+    };
+
+    [[maybe_unused]] auto copy_tau = [k, &tau, &taus](SizeType i, whip::stream_t stream) {
+      gpulapack::lacpy(blas::Uplo::General, k, 1, tau.ptr(), tau.ld(), taus(i).ptr(), taus(i).ld(),
+                       stream);
+    };
+
+    [[maybe_unused]] auto kernel_GPU1 = [m, k, &vs, &ts](SizeType i, cublasHandle_t handle) {
+      gpulapack::larft_gemv1_notau(handle, m, k, vs(i).ptr(), vs(i).ld(), ts(i).ptr(), ts(i).ld());
+    };
+
+    [[maybe_unused]] auto post_kernel_GPU1 = [m, k, &taus, &ts](SizeType i, whip::stream_t stream) {
+      gpulapack::larft_gemv1_fixtau(k, taus(i).ptr(), 1, ts(i).ptr(), ts(i).ld(), stream);
+    };
+#endif
+    const double flop = ops<T>(n, k);
+
+    KernelRunner<backend> runner(opts.count, opts.nparallel);
+
+    for (SizeType run_index = 0; run_index < opts.nruns; ++run_index) {
+      double elapsed_time_pre = 0;
+      double elapsed_time_kernel = 0;
+      double elapsed_time_post = 0;
+      ts.setElements(el_t);
+
+      if constexpr (backend == Backend::MC) {
+        elapsed_time_kernel = runner.run(kernel_MC);
+      }
+#ifdef DLAF_WITH_CUDA
+      if constexpr (backend == Backend::GPU) {
+        switch (opts.kernel_id) {
+          case 0:
+            elapsed_time_kernel = runner.runHandle(kernel_GPU0);
+            elapsed_time_post = runner.runStream(copy_tau_in_t);
+            break;
+          case 1:
+            elapsed_time_pre = runner.runStream(copy_tau);
+            elapsed_time_kernel = runner.runHandle(kernel_GPU1);
+            elapsed_time_post = runner.runStream(post_kernel_GPU1);
+            break;
+          default:
+            std::cout << "Error: Non existing kernel id" << opts.kernel_id << std::endl;
```

Comment:
```suggestion
            std::cout << "Error: Nonexistent kernel id" << opts.kernel_id << std::endl;
```
and then `std::terminate` or exit?

Reply:
https://github.com/eth-cscs/DLA-Future/pull/1219/files#diff-ddd8b6960bce12438abe598f5c25f1d5a2cddb794b5aeba723be222f9d717dddR160